WordPress/Classes/ViewRelated/Notifications/Views/NoteTableViewCell.swift

```diff
@@ -89,7 +89,14 @@ import WordPressShared
         
         // Handle non-gravatar images
         let placeholderImage = Style.blockGravatarPlaceholderImage(isApproved: !unapproved)
-        iconImageView.downloadImage(url, placeholderImage: placeholderImage)
+        iconImageView.downloadImage(url, placeholderImage: placeholderImage, success: nil, failure: { (error) in
+            // Note: Don't cache 404's. Otherwise Unapproved / Approved gravatars won't switch!
+            if self.gravatarURL?.isEqual(url) == true {
+                self.gravatarURL = nil
```

Comment:
No risk on this one, since the callback and `gravatarURL` assignment (few lines below) both happen on the Main Thread, always. Otherwise UIKit would get annoyed.

Thanks for checking!!